### ISSUE TITLE
Does or will sni.dll have a native version for Android?

### ISSUE THREAD

**Issue by akshays2112** [dotnet/SqlClient#748]
My dabbling seems to have pointed to me that sni.dll is native and not managed and hence platform-specific.

Is there a native sni.dll for Android?

If there is not native sni.dll for Android are there any plans to make one? If there are plans what is expected timeline to deliverable?

Thanks for your patience and time.

Best Regards,

Akshay Srinivasan

**Comment by JRahnama** [comment]
@akshays2112 There two types of SNI in our driver, as you have mentioned native SNI and managed SNI. Managed SNI are designed for netcore platform and are to be used on Unix or non-windows platforms such as Unix and etc., however you can use `Appcontext` switch on Windows and force using of managed SNI. You can read more about it [here](https://github.com/dotnet/SqlClient/blob/master/release-notes/2.0/2.0.0.md).

In other words, add `AppContext.SetSwitch("Switch.Microsoft.Data.SqlClient.UseManagedNetworkingOnWindows", true);` at the beginning of your application  and that's it. It will set using of `Managed SNI`  to true on Windows applications as well.

**Comment by cheenamalhotra** [comment]
Hi @akshays2112 

As @JRahnama mentioned, we already have a Managed SNI implementation compatible for Android applications.
Closing issue.
